### PR TITLE
New build process

### PR DESCRIPTION
**Pull request by rootfs**
- [x] (from Vimal) Use the new base and kepler images
- [x] (from Vimal) Upgrade to golang 1.20
- [x] Remove all bcc build from Makefile and GHA workflow

### CODE REVIEW

File: .github/workflows/image_base.yml [hunk 1]
```diff
@@ -57,9 +33,9 @@ jobs:
         uses: docker/build-push-action@v5
         with:
           context: .
-          file: ./Dockerfile.base
+          file: ./build/Dockerfile.builder
```
Comment (vimalk78) [inline .github/workflows/image_base.yml:36]: @rootfs I had added Dockerfiles to t he root of repo. If we want to keep Dockerfiles under `build` folder, pls move the `Dockerfile.builder` to `build` folder.
https://github.com/sustainable-computing-io/kepler/pull/1118/files#diff-e7cd9380abeae311f2b7f9ed2e780a7ea795597555411cc965191ca1e6cedf16

Comment (rootfs) [reply .github/workflows/image_base.yml:36]: The file is copied there, I'll remove it

File: .github/workflows/image_base.yml [hunk 2]
```diff
@@ -1,9 +1,7 @@
 name: base
 
 on:
-  push:
-    branches: 
-      - 'base'
+  workflow_dispatch:
```
Comment (vimalk78) [inline .github/workflows/image_base.yml:4]: base image build needs to be triggered manually now?

File: .github/workflows/image_base.yml (hunk 1 above)
Comment (vimalk78) [reply .github/workflows/image_base.yml:26]: can you point which GH workflow step copies the `Dockerfile.builder` ?

**Review by vimalk78 — COMMENTED** [review]
if we are removing `bcc` altogether, we can (perhaps in another PR) remove files which build with `bcc` flag
https://github.com/sustainable-computing-io/kepler/blob/main/pkg/bpfassets/attacher/bcc_attacher.go#L1-L2
https://github.com/sustainable-computing-io/kepler/blob/main/pkg/bpfassets/attacher/attacher_suite_test.go#L1-L2
https://github.com/sustainable-computing-io/kepler/blob/main/e2e/platform-validation/platform_validation_metric_test.go#L1-L2
https://github.com/sustainable-computing-io/kepler/blob/main/e2e/integration-test/e2e_metric_test.go#L1-L2

**Comment by rootfs** [comment]
@vimalk78 For now we just remove the bcc image from build and release so we don't have to release and maintain them. The entire bcc stack will removed later.

File: .github/workflows/image_base.yml (hunk 2 above)
Comment (rootfs) [reply .github/workflows/image_base.yml:4]: the multi arch image build is still not working. Let's create a separate PR for that. For now, I just made this a manual workflow.

Comment (vimalk78) [reply .github/workflows/image_base.yml:4]: > the multi arch image build is still not working

can you point me to the pipeline?

Comment (rootfs) [reply .github/workflows/image_base.yml:4]: https://github.com/sustainable-computing-io/kepler/blob/main/.github/workflows/image_base.yml

**Comment by vprashar2929** [comment]
Tested the image built from the new `dockerfile` on OpenShift and it works fine.

**Comment by rootfs** [comment]
thank you @vprashar2929

**Review by husky-parul — APPROVED** [review]

**Pull request merged**